components/brave_wallet/browser/brave_wallet_utils.cc

```diff
@@ -385,4 +386,99 @@ void UpdateLastUnlockPref(PrefService* prefs) {
   prefs->SetTime(kBraveWalletLastUnlockTime, base::Time::Now());
 }
 
+base::Value TransactionReceiptToValue(const TransactionReceipt& tx_receipt) {
+  base::Value dict(base::Value::Type::DICTIONARY);
+  dict.SetStringKey("transaction_hash", tx_receipt.transaction_hash);
+  dict.SetStringKey("transaction_index",
+                    Uint256ValueToHex(tx_receipt.transaction_index));
+  dict.SetStringKey("block_hash", tx_receipt.block_hash);
+  dict.SetStringKey("block_number", Uint256ValueToHex(tx_receipt.block_number));
+  dict.SetStringKey("from", tx_receipt.from);
+  dict.SetStringKey("to", tx_receipt.to);
+  dict.SetStringKey("cumulative_gas_used",
+                    Uint256ValueToHex(tx_receipt.cumulative_gas_used));
+  dict.SetStringKey("gas_used", Uint256ValueToHex(tx_receipt.gas_used));
+  dict.SetStringKey("contract_address", tx_receipt.contract_address);
+  // TODO(darkdh): logs
+  dict.SetStringKey("logs_bloom", tx_receipt.logs_bloom);
+  dict.SetBoolKey("status", tx_receipt.status);
+  return dict;
+}
+
+base::Optional<TransactionReceipt> ValueToTransactionReceipt(
```

Comment:
I just noticed this when rebasing the cr93 branch, so thought I'd comment even if this has been merged:

Please avoid using the deprecated `base::Optional` and `base::nullopt` in new code, and use `absl::optional` and `absl::nullopt` instead. That will help with rebases, at least with the cr93 one :innocent: , thanks!

There are 20 instances of the former and 48 of the later in this commit, I'll prepare a PR for master now if that's ok